src/XRControllerModel.ts

```diff
@@ -15,11 +15,19 @@ const isEnvMapApplicable = (
   material: any
 ): material is MeshBasicMaterial | MeshStandardMaterial | MeshPhongMaterial | MeshLambertMaterial => 'envMap' in material
 
-const applyEnvironmentMap = (envMap: Texture, envMapIntensity: number, obj: Object3D): void => {
+const applyEnvironmentMap = (envMap: Texture | null, obj: Object3D): void => {
   obj.traverse((child) => {
     if (child instanceof Mesh && isEnvMapApplicable(child.material)) {
       child.material.envMap = envMap
-      if ('envMapIntensity' in child.material) child.material.envMapIntensity = envMapIntensity
+      child.material.needsUpdate = true
+    }
+  })
+}
+
+const applyEnvironmentMapIntensity = (envMapIntensity: number, obj: Object3D): void => {
+  obj.traverse((child) => {
+    if (child instanceof Mesh && 'envMapIntensity' in child.material) {
```

Comment:
This should narrow with `child instanceof Mesh && child.material instanceof Material && 'envMapIntensity' in child.material` for the array materials case.

Reply:
I'd say it should handle array materials also, in this case

Reply:
Supported array materials